.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,240 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+#
+# See https://github.com/r-lib/actions/tree/master/examples#readme for
+# additional example workflows available for the R community.
+
+name: R
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.config.os-name }}-${{ matrix.config.os-version }}
+    name: >- 
+      ${{ matrix.config.os-name }}-${{ matrix.config.os-version }}
+      R ${{ matrix.config.r-version}} - java ${{ matrix.config.java}}
+    strategy:
+      matrix:
+        config: 
+          - os-name: ubuntu # windows-2019, macos-10.15]
+            os-version: latest
+            r-version: oldrel
+            java: 15
+            vignettes: false
+            xlc-repo: https://jcenter.bintray.com
+          - os-name: ubuntu # windows-2019, macos-10.15]
+            os-version: "18.04"
+            r-version: release
+            java: 11
+            vignettes: true
+            timezone-name: Asia/Kathmandu
+          - os-name: ubuntu # windows-2019, macos-10.15]
+            os-version: "18.04"
+            r-version: devel
+            java: 8
+            vignettes: false
+            xlc-repo: https://jcenter.bintray.com
+            timezone-name: Pacific/Marquesas
```

Comment:
I wanted to have "weird" combinations as it seems that would be more likely to reveal issues. The macOs and windows ones I would consider more standard.

Reply:
Re: timezones, we had a previous bug in XLConnect where timezones played a role, which is why I'm using different timezones (esp. weird ones with quarter-hour offsets) - these were present in travis-ci already. 
I didn't want to have too many combinations. Maybe this can be reconsidered in a future PR, since github minutes are unlimited for OSS projects ? It still would make the overall build take more time.

Reply:
Opened #119 as follow-up.